File: src/strategy/examples/searchintent/SaleSearchStrategy.java
package strategy.examples.searchintent;

import java.util.List;

import abstractfactory.examples.searchcatalog.CatalogFactory;
import adapter.examples.metadata.ImageMetadata;

/** RF2.4: sale -> only certified breeders. */
public class SaleSearchStrategy implements SearchStrategy {

    private final CatalogFactory factory;

    public SaleSearchStrategy(CatalogFactory factory) { this.factory = factory; }

    @Override
    public SearchResult search(ImageMetadata meta) {
        List<String> matches = factory.createBreederCatalog().query(meta);
        return new SearchResult(SearchIntent.SALE, matches);
    }
}

File: src/strategy/examples/searchintent/SaleSearchStrategy.js
'use strict';

const SearchIntent = require('./SearchIntent');
const SearchResult = require('./SearchResult');

/** RF2.4: venta -> solo criaderos certificados. */
class SaleSearchStrategy {
  constructor(factory) { this.factory = factory; }
  search(meta) {
    const matches = this.factory.createBreederCatalog().query(meta);
    return new SearchResult(SearchIntent.SALE, matches);
  }
}

module.exports = SaleSearchStrategy;
